custom/icds/location_reassignment/tasks.py

```diff
@@ -1,8 +1,31 @@
+from django.conf import settings
+from django.core.mail.message import EmailMessage
+
 from celery.task import task
 
 from custom.icds.location_reassignment.processor import Processor
 
 
 @task
-def process_location_reassignment(domain, transitions, site_codes):
-    Processor(domain, transitions, site_codes).process()
+def process_location_reassignment(domain, transitions, site_codes, user_email):
+    try:
+        Processor(domain, transitions, site_codes).process()
+    except Exception as e:
+        email = EmailMessage(
+            subject='[{}] - Location Reassignment Failed'.format(settings.SERVER_ENVIRONMENT),
+            body="The request could not be completed. Something went wrong. "
+                 "Error raised : {}. "
+                 "Please report an issue if needed.".format(e),
+            to=[user_email],
+            from_email=settings.DEFAULT_FROM_EMAIL
+        )
+        email.send()
+        raise e
+    else:
```

Comment:
Is there a reason to prefer this being in an `else` versus just after the try/except?

Reply:
I like keeping what would cause the exception in between try and except so its clear where it would come from.